### PR TITLE
Add StopCaptureReason::kExceededMaxDurationLimit

### PR DESCRIPTION
**Pull request by vickyliu-go4it**
The purpose is to enable the CloudCollector to limit the capture length by a fixed time duration.

In this change, we enable the `CloudCollectorStartStopCaptureRequestWaiter` to stop waiting for 
the external stop capture request and stop the capture internally if the capture duration exceeds 
the maximum capture duration limit, e.g., 10 min.
Also a `StopCaptureReason::kExceededMaxDurationLimit` and the corresponding `CaptureFinishedEvent` 
are added for this case.

Bug: http://b/222681438

### CODE REVIEW

File: src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp [hunk 1]
```diff
@@ -27,7 +28,14 @@ void CloudCollectorStartStopCaptureRequestWaiter::StartCapture(CaptureOptions ca
 CaptureServiceBase::StopCaptureReason
 CloudCollectorStartStopCaptureRequestWaiter::WaitForStopCaptureRequest() {
   absl::MutexLock lock(&stop_mutex_);
-  stop_mutex_.Await(absl::Condition(&stop_requested_));
+
+  if (max_capture_duration_.has_value()) {
```
Comment (beckerhe) [inline src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp:32]: Isn't that the wrong way around? "If there is **NO** max_capture_duration set, then we wait without a deadline", right? What am I missing? :sweat_smile:

File: src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp [hunk 2]
```diff
@@ -27,7 +28,14 @@ void CloudCollectorStartStopCaptureRequestWaiter::StartCapture(CaptureOptions ca
...
+    stop_mutex_.Await(absl::Condition(&stop_requested_));
+  } else if (!stop_mutex_.AwaitWithDeadline(absl::Condition(&stop_requested_),
```
Comment (beckerhe) [inline src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp:34]: Could you return early in the previous case and such avoid the else?

**Review by beckerhe — APPROVED** [review]
Thank you Vicky. Otherwise LGTM

File: src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp (hunk 1 above)
Comment (vickyliu-go4it) [reply src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp:32]: If no max_capture_duration set, we will wait for the external stop request, i.e., someone call `CloudCollectorStartStopCaptureRequestWaiter::StopCapture`

File: src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp [hunk 3]
```diff
@@ -13,6 +13,7 @@ namespace orbit_capture_service_base {
 CaptureOptions CloudCollectorStartStopCaptureRequestWaiter::WaitForStartCaptureRequest() {
   absl::MutexLock lock(&start_mutex_);
   start_mutex_.Await(absl::Condition(&start_requested_));
+  capture_start_time_ = absl::Now();
```
Comment (dpallotti) [inline src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp:16]: A possible simplification is to just use `AwaitWithTimeout` in `WaitForStopCaptureRequest` below and get rid of this field (which might also be a bit confusing as we have a similar `capture_start_unix_time_ns` in `CaptureStarted`). After all you don't need to be too precise and the only difference is between including the capture initialization time or not.

**Review by dpallotti — APPROVED** [review]

File: src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp (hunk 3 above)
Comment (vickyliu-go4it) [reply src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp:16]: Thanks for the suggestion. But then the `CloudCollectorStartStopCaptureRequestWaiter` needs to access the CaptureStarted event in `WaitForStopCaptureRequest`.

**Pull request merged**